uber/templates/registration/view_promo_codes.html

```diff
@@ -0,0 +1,230 @@
+{% extends "base-admin.html" %}
+{% block title %}View Promo Codes{% endblock %}
+{% block content %}
+<style>
+.decent > th,td{
+    padding: 1em;
+}
+.decent-rows:hover{
+    background-color: #f5f5f5;
+}
+.delete{
+    font-size: 36px;
+}
+
+.delete:hover{
+    color: #FF0011;
+}
+</style>
+<h2> View Promo Codes </h2>
+
+<table>
+<tr class="decent">
+    <th>
+        Delete?
+    </th>
+    <th>
+        Code
+    </th>
+    <th>
+        Expiration Date
+    </th>
+    <th>
+        Uses
+    </th>
+    <th>
+        Ticket Price
+    </th>
+    <th>
+        Expired
+    </th>
+    <th>
+        Send Email?
+    </th>
+    <th>
+        Users
+    </th>
+</tr>
+{% for p in promo_codes %}
+    <tr class="decent decent-rows">
+        <td class="delete">
+            <a href="#" onclick="deleteCode('{{ p.id }}')"><i>X</i></a>
+        </td>
+        <td>
+            <a href="#" onclick="editCode('{{ p.id }}', '{{ p.code }}')">{{ p.code }}</a>
+        </td>
+        <td>
+            <a href="#" id="date{{ forloop.counter }}" name="date{{ forloop.counter }}" onclick="dateID = '{{ p.id }}'; cal.showCalendar('date{{ forloop.counter }}')"><i>{{ p.expiration_date }}</i></a>
+        </td>
+        <td>
+            {% if p.uses == None %}
+                Unlimited
+            {% else %}
+                {{ p.uses }}
+            {% endif %}
+        </td>
+        <td>
+            ${{ p.price }}
+        </td>
+        <td>
+            {% if not p.expired %}<a href="#" onclick="expireCode('{{ p.id }}')">False</a>{% else %}True{% endif %}
+        </td>
+        <td>
+            <form onsubmit="return generateEmail(this)" id="emailForm{{ forloop.counter }}" method="POST"
```

Comment:
Again, we have much nicer and more user-friendly modal pop-ups for this. The default JavaScript prompts you used are really to be avoided due to various usability issues. For example, if I click "Cancel" on the email prompt, it stills asks me for a name. That's just sill.